src/ucp/wireup/wireup.c

```diff
@@ -202,29 +192,111 @@ static ucs_status_t ucp_wireup_msg_send(ucp_ep_h ep, uint8_t type,
 
     req->send.buffer = address;
 
-    /* send the indices addresses that should be connected by remote side */
+    ucp_request_send(req, 0);
+    return UCS_OK;
+}
+
+static uint64_t ucp_wireup_get_ep_tl_bitmap(ucp_ep_h ep, ucp_lane_map_t lane_map)
+{
+    uint64_t         tl_bitmap = 0;
+    ucp_lane_index_t lane;
+
+    ucs_for_each_bit(lane, lane_map) {
+        ucs_assert(lane < UCP_MAX_LANES);
+        tl_bitmap |= UCS_BIT(ucp_ep_get_rsc_index(ep, lane));
+    }
+
+    return tl_bitmap;
+}
+
+/*
+ * Select remote ep address for every remote address entry (because there
+ * could be multiple ep addresses per entry). This selection is used to create
+ * 'lanes2remote' mapping with the remote lane index for each local lane.
+ */
+static void
+ucp_wireup_match_p2p_lanes(ucp_ep_h ep,
+                           const ucp_unpacked_address_t *remote_address,
+                           const unsigned *addr_indices,
+                           ucp_lane_index_t *lanes2remote)
+{
+    const ucp_address_entry_t *address;
+    unsigned address_index;
+    ucp_lane_index_t lane, remote_lane;
+    unsigned *ep_addr_indexes;
+    unsigned ep_addr_index;
+    uint64_t UCS_V_UNUSED used_remote_lanes;
+
+    /* Initialize the counters of ep address index for each address entry */
+    ep_addr_indexes = ucs_alloca(sizeof(ep_addr_index) *
+                                 remote_address->address_count);
+    for (address_index = 0; address_index < remote_address->address_count;
+         ++address_index) {
+        ep_addr_indexes[address_index] = 0;
```

Comment:
memset?

Reply:
i prefered for because the entry type is 'unsigned', it's not just filling a buffer